src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -396,10 +396,23 @@ private List<Feedback> saveFeedbackWithHibernateWorkaround(@NotNull Result resul
         feedbackList.forEach(feedback -> {
             // cut association to parent object
             feedback.setResult(null);
+            LongFeedbackText longFeedback = null;
+            // search for long feedback parent feedback has and cut the association between them.
+            if (feedback.getId() != null) {
+                longFeedback = longFeedbackTextRepository.findByFeedbackId(feedback.getId()).orElse(null);
```

Comment:
Would it be possible to reduce this to a single database-query?
Before the start of the `feedbackList.forEach` loop, you could maybe do something like
```java
List<Long> feedbackIdsWithLongFeedback = feedbackList
    .stream()
    .filter(feedback -> feedback.id != null && feedback.hasLongFeedbackText())
    .map(feedback -> feedback.id)
    .toList();
```
After that it should be possible to load all related long feedbacks with a new method/query in the repository somewhat like
```sql
select lf
from LongFeedbackText lf
where lf.feedback_id in :feedbackIds
```

Then you should have a list of all required long feedback texts to connect to the feedback again here.